Add option to record redact() SDK requests (#10)
TonicAI/textual: docs/source/redact/redacting_text.rst

        "new_text": "[ORGANIZATION_P5XLAH]"
    }

> `redact` calls can also be recorded so you can view and analyze results in our UI.  To learn more, read :ref:`record-api-call-section`

Reviewer: ```suggestion
You can also record `redact` calls, so that you can view and analyze results in the Textual application. To learn more, read :ref:`record-api-call-section`
```

    }

Note that LLM Synthesis is non-deterministic — you will likely get different results each time you run it.

.. _record-api-call-section:

Recording API requests
----------------------

> When you redact text with the :meth:`redact<tonic_textual.redact_api.TextualNer.redact>` method you can optionally record these requests for later viewing and analysis in the UI.  The `redact` method takes an optional argument called `record_options` (:class:`RecordApiRequestOptions<tonic_textual.classes.record_api_request_options.RecordApiRequestOptions>`).  To record an API request you can do the following:

Reviewer: ```suggestion
When you use the :meth:`redact<tonic_textual.redact_api.TextualNer.redact>` method to redact text, you can optionally record these requests to view and analyze later in the Textual application. The `redact` method takes an optional `record_options` (:class:`RecordApiRequestOptions<tonic_textual.classes.record_api_request_options.RecordApiRequestOptions>`) argument. To record an API request:
```

.. code-block:: python

    from tonic_textual.redact_api import TextualNer
    from tonic_textual.classes.record_api_request_options import RecordApiRequestOptions

    ner = TextualNer()

    ner.redact("My name is John Doe", record_options=RecordApiRequestOptions(
        record=True,
        retention_time_in_hours=1,
        tags=["my_first_request"])
    )

> The above code will run the redaction as normal, and then record the API Request and its results.  The request itself will be automatically purged after 1 hour.  Results are viewable in the UI from the API Explorer page.

Reviewer: ```suggestion
The above code runs the redaction in the same way as any other redaction request, and then records the API request and its results. The request itself is automatically purged after 1 hour.  You can view the results from the **API Explorer** page in Textual.
```
